### ISSUE TITLE
running preprocess.py occurs NAN

### ISSUE THREAD

**Issue by runningJ** [xcmyz/FastSpeech#50]
print(mel_outputs)
None
tensor([[[nan, nan, nan,  ..., nan, nan, nan],
         [nan, nan, nan,  ..., nan, nan, nan],
         [nan, nan, nan,  ..., nan, nan, nan],
         ...,
         [nan, nan, nan,  ..., nan, nan, nan],
         [nan, nan, nan,  ..., nan, nan, nan],
         [nan, nan, nan,  ..., nan, nan, nan]]], device='cuda:0')

**Comment by xcmyz** [comment]
Could you provide more information?

**Comment by runningJ** [comment]
I dowanload the pretrained model from nvidia. My pytorch version is 1.2. when i run pereprocess.py occurs this problem.

`def forward(self, inputs):
        text_inputs, text_lengths, mels, max_len, output_lengths = inputs
        text_lengths, output_lengths = text_lengths.data, output_lengths.data

        embedded_inputs = self.embedding(text_inputs).transpose(1, 2)

        encoder_outputs = self.encoder(embedded_inputs, text_lengths)

        mel_outputs, gate_outputs, alignments = self.decoder(
            encoder_outputs, mels, memory_lengths=text_lengths)

        mel_outputs_postnet = self.postnet(mel_outputs)
        mel_outputs_postnet = mel_outputs + mel_outputs_postnet`

encoder_outputs has value.

**Comment by xcmyz** [comment]
> I dowanload the pretrained model from nvidia. My pytorch version is 1.2. when i run pereprocess.py occurs this problem.
> 
> `def forward(self, inputs):
> text_inputs, text_lengths, mels, max_len, output_lengths = inputs
> text_lengths, output_lengths = text_lengths.data, output_lengths.data
> 
> ```
>     embedded_inputs = self.embedding(text_inputs).transpose(1, 2)
> 
>     encoder_outputs = self.encoder(embedded_inputs, text_lengths)
> 
>     mel_outputs, gate_outputs, alignments = self.decoder(
>         encoder_outputs, mels, memory_lengths=text_lengths)
> 
>     mel_outputs_postnet = self.postnet(mel_outputs)
>     mel_outputs_postnet = mel_outputs + mel_outputs_postnet`
> ```
> 
> encoder_outputs has value.

Maybe you should use Pytorch 1.1.0.

**Comment by runningJ** [comment]
I solved the problem by modifying the following code.

`def get_mask_from_lengths(lengths):
    max_len = torch.max(lengths).item()
    ids = torch.arange(0, max_len, out=torch.cuda.LongTensor(max_len))
    mask = (ids < lengths.unsqueeze(1)).bool()
    return mask`
